### PR TITLE
Allow to edit restriction form during creation workflow

### PR DESCRIPTION
**Pull request by phiz71**
## Issue

https://gravitee.atlassian.net/browse/APIM-1969

## Description

Allow the edition of restrictions of a plan during the API creation workflow.

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-hteezlmefx.chromatic.com)

### CODE REVIEW

**Comment by jourdiw** [comment]
There are two commits on this branch with the same name... Maybe squash the two?

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-4-security/step-4-security-1-plans.harness.ts [hunk 1]
```diff
@@ -121,11 +122,12 @@ export class Step4Security1PlansHarness extends ComponentHarness {
...
 
     await (await this.getButtonByText('Next')).click();
-
     const rateLimitToggle = await apiPlanFormHarness.getRateLimitEnabledInput();
     await rateLimitToggle.toggle();
 
-    apiPlanFormHarness.httpRequest(httpTestingController).expectPolicySchemaGetRequest('rate-limit', {});
+    // When form has initial value, value is patched and multiple requests can be sent to get the schema. Need to use match here, but we check that at least one request is done
+    const req = httpTestingController.match({ url: `${CONSTANTS_TESTING.env.baseURL}/policies/rate-limit/schema`, method: 'GET' });
+    expect(req.length > 0);
```
Comment (phiz71) [inline gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-4-security/step-4-security-1-plans.harness.ts]: I remember I had the same issue on another test. I realized that I could ignore first requests as they were "cancelled".
https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-console-webui/src/management/api/general/subscriptions/list/api-portal-subscription-list.component.spec.ts#L846-L854

Maybe it is the same for you here?

**Review by phiz71 — COMMENTED** [review]
LGTM. Can't approve since I've opened this PR 😛 .

File: gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-4-security/step-4-security-1-plans.harness.ts (hunk 1 above)
Comment (jourdiw) [reply gravitee-apim-console-webui/src/management/api/creation-v4/steps/step-4-security/step-4-security-1-plans.harness.ts]: You can also configure the httpTestController to ignore cancelled calls if that's the case

**Review by jourdiw — APPROVED** [review]

**Pull request merged**